test/e2e/e2e_test.go

```diff
@@ -19,192 +19,116 @@ package e2e
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
-	"path"
+	"os"
+	"path/filepath"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"k8s.io/apimachinery/pkg/types"
 
-	infrav1 "sigs.k8s.io/cluster-api-provider-digitalocean/api/v1alpha3"
-	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
 
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
-	bootstrapkubeadmv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	capie2e "sigs.k8s.io/cluster-api/test/e2e"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
 	"sigs.k8s.io/cluster-api/util"
-	"sigs.k8s.io/cluster-api/util/kubeconfig"
 )
 
-var _ = Describe("functional tests", func() {
-	Describe("cluster lifecycle", func() {
-		var (
-			clusterName                string
-			clusterNamespace           string
-			clusterGenenerator         ClusterGenerator
-			machineGenerator           MachineGenerator
-			machineDeploymentGenerator MachineDeploymentGenerator
-		)
-
-		BeforeEach(func() {
-			var err error
-			clusterName = "capdo-test-" + util.RandomString(6)
-			clusterNamespace = "default"
-
-			testTmpDir, err = ioutil.TempDir(suiteTmpDir, "e2e-test")
-			Expect(err).NotTo(HaveOccurred())
+var _ = Describe("Cluster Creation", func() {
```

Comment:
This test actually just like [capi e2e quickstart](https://pkg.go.dev/sigs.k8s.io/cluster-api@v0.3.9/test/e2e#QuickStartSpec). I extract the function here because we need to change the clusterctl log folder to somewhere (other than artifacts folder) since our credentials (in digitalocean cloud-controller-manager manifest) exposed there.